test/e2e/kubectl/debug.go

```diff
@@ -0,0 +1,188 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// OWNER = sig/cli
+
+package kubectl
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+	testutils "k8s.io/kubernetes/test/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("kubectl debug", func() {
+	defer ginkgo.GinkgoRecover()
+	f := framework.NewDefaultFramework("kubectl-debug")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
+
+	var ns string
+	var c clientset.Interface
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+	})
+
+	ginkgo.Describe("custom profile", func() {
+		ginkgo.It("should be applied on static profiles on ephemeral container", func(ctx context.Context) {
+			debugPodName := "e2e-test-debug-custom-profile-pod-ephemeral"
+			customProfileImage := imageutils.GetE2EImage(imageutils.BusyBox)
+			ginkgo.By("running the image " + customProfileImage)
+			e2ekubectl.RunKubectlOrDie(ns,
+				"run", debugPodName,
+				"--image="+customProfileImage,
+				podRunningTimeoutArg,
+				"--labels=run="+debugPodName,
+				"--", "sh", "-c",
+				"sleep 3600")
+
+			ginkgo.By("verifying the pod " + debugPodName + " is running")
+			label := labels.SelectorFromSet(map[string]string{"run": debugPodName})
+			err := testutils.WaitForPodsWithLabelRunning(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed getting pod %s: %v", debugPodName, err)
+			}
+
+			tmpDir, err := os.MkdirTemp("", "test-custom-profile-debug")
+			customProfileFile := "custom.yaml"
+			framework.ExpectNoError(err)
+			defer os.Remove(tmpDir) //nolint:errcheck
+			framework.ExpectNoError(os.WriteFile(filepath.Join(tmpDir, customProfileFile), []byte(`
+securityContext:
+  capabilities:
+    add:
+      - NET_ADMIN
+env:
+  - name: REQUIRED_ENV_VAR
+    value: value2
+`), os.FileMode(0755)))
+			ginkgo.By("verifying ephemeral container has correct fields")
+			e2ekubectl.RunKubectlOrDie(ns,
+				"debug",
+				debugPodName,
+				"--image="+imageutils.GetE2EImage(imageutils.Nginx),
+				"--container=debugger",
+				"--profile=general", // general profile sets capability to SYS_PTRACE and custom should overwrite it to NET_ADMIN
+				"--custom="+filepath.Join(tmpDir, customProfileFile),
+				"--", "sh", "-c",
+				"sleep 3600")
+
+			ginkgo.By("verifying the container debugger is running")
+			err = wait.PollUntilContextTimeout(ctx, 3*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
```

Comment:
How about using `WaitForPodContainerStarted` from `k8s.io/kubernetes/test/e2e/framework/pod`?

Reply:
After your suggestion, I've found a better one `WaitForPodsWithLabelRunning` since I'm focusing on the `running` status